comp/core/tagger/tagger.go

```diff
@@ -365,8 +367,14 @@ func (t *TaggerClient) ResetCaptureTagger() {
 func (t *TaggerClient) EnrichTags(tb tagset.TagsAccumulator, originInfo taggertypes.OriginInfo) {
 	cardinality := taggerCardinality(originInfo.Cardinality)
 
+	// If dogstatsdUnifiedEnabled is disabled, we use DogStatsD's Legacy Origin Detection.
+	// TODO: remove this when dogstatsd_origin_detection_unified is enabled by default
+	if !t.datadogConfig.originDetectionUnifiedEnabled {
+		originInfo.ProductOrigin = taggertypes.ProductOriginDogStatsDLegacy
+	}
```

Comment:
a cleaner way would be to consider that `originInfo` is const
and instead of overriding it a local var is use to set get the ProductOrigin

```suggestion
    productOrigin := originInfo.ProductOrigin
	if !t.datadogConfig.originDetectionUnifiedEnabled {
		productOrigin = taggertypes.ProductOriginDogStatsDLegacy
	}
```